### PR TITLE
chore(#8306): remove grunt

### PR DESCRIPTION
**Pull request by garethbowen**
# Description

medic/cht-core#8306

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:remove-grunt/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:remove-grunt/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:remove-grunt/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by garethbowen** [comment]
@dianabarsan Would you mind reviewing? This is clearly not complete but I don't know when I'll get the chance to make another push so thought it was worth committing progress to date. For now I'm making heavy use of grunt-exec which ultimately will become npm scripts but it's a lighter lift for now to keep everything in grunt before moving it all over in a final step.

File: Gruntfile.js [hunk 1]
```diff
@@ -28,51 +26,14 @@ module.exports = function(grunt) {
...
-      testing: {
-        files: [
-          {
-            src: 'build/staging.json',
-            dest: `${MARKET_URL}/${BUILDS_SERVER}`,
-          },
-        ],
-      }
-    },
+    // this probably needs a script - can't find an config file option
```
Comment (dianabarsan) [inline Gruntfile.js:36]: I believe you added the script, right?

**Review by dianabarsan — APPROVED** [review]
Cool stuff! The only thing I don't like about npm commands is that they are executed in an isolated env scope, so passing env from one to another is not possible.

File: Gruntfile.js (hunk 1 above)
Comment (garethbowen) [reply Gruntfile.js:36]: For browserify? Not that I recall?

File: scripts/build/ddoc-compile.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
+const util = require('util');
+const couchCompile = util.promisify(require('couchdb-compile'));
+const { writeFile } = require('node:fs/promises');
+
+const compilePrimary = async () => {
+  await compile(
+    [ // todo: scan directory?
```
[the remark this answers is not in the document]
Comment (garethbowen) [reply scripts/build/ddoc-compile.js]: Done.

File: scripts/build/push-ddoc-to-staging.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+const { readFile } = require('node:fs/promises');
+const PouchDB = require('pouchdb-core');
+PouchDB.plugin(require('pouchdb-adapter-http'));
+const { MARKET_URL, BUILDS_SERVER } = process.env;
+const URL = `${MARKET_URL}/${BUILDS_SERVER}`;
+
+(async () => {
+  const ddocsBuffer = await readFile('build/staging.json');
+  const ddocs = JSON.parse(ddocsBuffer.toString());
+  const ddoc = ddocs && ddocs.docs && ddocs.docs.length && ddocs.docs[0];
```
[the remark this answers is not in the document]
Comment (garethbowen) [reply scripts/build/push-ddoc-to-staging.js]: Done.

**Pull request merged**